pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -128,9 +133,7 @@ func (c *Controller) Apply(change *sink.Change) error {
 		return fmt.Errorf("descriptor type not supported %s", change.Collection)
 	}
 
-	c.syncedMu.Lock()
-	c.synced[change.Collection] = true
-	c.syncedMu.Unlock()
+	c.sync(change.Collection)
 
 	// innerStore is [namespace][name]
 	innerStore := make(map[string]map[string]*model.Config)
```

Comment:
I think this entire method has to behave differently for incremental vs non-incremental.

For non-incremental (i.e. the current behavior): the snapshot contains all of the current resources.  We don't really care about which were removed, since we end up overwriting the old map with the new one.

For incremental: we will receive only the resources that have changed. In this case, we need to start with a copy of the old map and then modify it based on both the included resources as well as those that have been removed.

Reply:
This will change too once I update the PR as per our discussion.